lte/gateway/c/connection_tracker/EventTracker.cpp

```diff
@@ -0,0 +1,274 @@
+  const struct nlattr** tb = (const struct nlattr**) data;
+  int type                 = mnl_attr_get_type(attr);
+
+  if (mnl_attr_type_valid(attr, CTA_TUPLE_MAX) < 0) return MNL_CB_OK;
+
+  switch (type) {
+    case CTA_TUPLE_IP:
+      if (mnl_attr_validate(attr, MNL_TYPE_NESTED) < 0) {
+        perror("mnl_attr_validate");
+        return MNL_CB_ERROR;
+      }
+      break;
+    case CTA_TUPLE_PROTO:
+      if (mnl_attr_validate(attr, MNL_TYPE_NESTED) < 0) {
+        perror("mnl_attr_validate");
+        return MNL_CB_ERROR;
+      }
+      break;
+  }
+  tb[type] = attr;
+  return MNL_CB_OK;
+}
+
+static void print_tuple(
+    const struct nlattr* nest, struct flow_information* flow) {
+  struct nlattr* tb[CTA_TUPLE_MAX + 1] = {};
+
+  mnl_attr_parse_nested(nest, parse_tuple_cb, tb);
+  if (tb[CTA_TUPLE_IP]) {
+    parse_ip(tb[CTA_TUPLE_IP], flow);
+  }
+  if (tb[CTA_TUPLE_PROTO]) {
+    parse_proto(tb[CTA_TUPLE_PROTO], flow);
+  }
+}
+
+static int data_attr_cb(const struct nlattr* attr, void* data) {
+  const struct nlattr** tb = (const struct nlattr**) data;
+  int type                 = mnl_attr_get_type(attr);
+
+  if (mnl_attr_type_valid(attr, CTA_MAX) < 0) return MNL_CB_OK;
+
+  switch (type) {
+    case CTA_TUPLE_ORIG:
+      if (mnl_attr_validate(attr, MNL_TYPE_NESTED) < 0) {
+        perror("mnl_attr_validate");
+        return MNL_CB_ERROR;
+      }
+      break;
+    case CTA_TIMEOUT:
+    case CTA_MARK:
+    case CTA_SECMARK:
+      if (mnl_attr_validate(attr, MNL_TYPE_U32) < 0) {
+        perror("mnl_attr_validate");
+        return MNL_CB_ERROR;
+      }
+      break;
+  }
+  tb[type] = attr;
+  return MNL_CB_OK;
+}
+
+static int data_cb(const struct nlmsghdr* nlh, void* data) {
+  struct nlattr* tb[CTA_MAX + 1] = {};
+  struct nfgenmsg* nfg = (struct nfgenmsg*) mnl_nlmsg_get_payload(nlh);
+  struct flow_information flow;
+  struct in_addr src_ip;
+  struct in_addr dst_ip;
+
+  if ((nlh->nlmsg_type & 0xFF) != IPCTNL_MSG_CT_DELETE) {
+    return 0;
+  }
+
+  mnl_attr_parse(nlh, sizeof(*nfg), data_attr_cb, tb);
+  if (tb[CTA_TUPLE_ORIG]) {
+    print_tuple(tb[CTA_TUPLE_ORIG], &flow);
+  }
+
+  src_ip.s_addr = flow.saddr;
+  dst_ip.s_addr = flow.daddr;
+
+  switch (nlh->nlmsg_type & 0xFF) {
+    case IPCTNL_MSG_CT_NEW:
+      if (nlh->nlmsg_flags & (NLM_F_CREATE | NLM_F_EXCL))
+        MLOG(MINFO) << "     [NEW] src=" << inet_ntoa(src_ip) << ":"
+                    << ntohs(flow.sport) << " dst=" << inet_ntoa(dst_ip) << ":"
+                    << ntohs(flow.dport) << " proto=" << flow.l4_proto;
+      else
+        printf("%9s ", "[UPDATE] \n");
+      break;
+    case IPCTNL_MSG_CT_DELETE:
+      MLOG(MINFO) << "[DESTROY] src=" << inet_ntoa(src_ip) << ":"
+                  << ntohs(flow.sport) << " dst=" << inet_ntoa(dst_ip) << ":"
+                  << ntohs(flow.dport) << " proto=" << flow.l4_proto;
+      break;
+  }
+
+  if (tb[CTA_MARK]) {
+    MLOG(MINFO) << "From zone " << mnl_attr_get_u16(tb[CTA_ZONE]);
+  }
```

Comment:
Can we use this zoneid from pipelieD to filter out unnecessary conntrack events. the zone id can be passed by .yml file.

Reply:
Yeah its the plan, currently tb[CTA_MARK] is unset for some reason, I can't figure out why :|

Reply:
As I can validate basic functionality without this, I'd preffer to leave it as a todo